service/datastore/client-rest/src/main/java/org/eclipse/kapua/service/datastore/client/rest/RestDatastoreClient.java

```diff
@@ -0,0 +1,641 @@
+            synchronized (RestDatastoreClient.class) {
+                if (instance == null) {
+                    if (esClientProvider != null) {
+                        cleanupClient(true);
+                    }
+                    logger.info("Starting Elasticsearch rest client...");
+                    esClientProvider = EsRestClientProvider.init();
+                    logger.info("Starting Elasticsearch rest client... DONE");
+                }
+            }
+        }
+    }
+
+    @Override
+    public void close() throws ClientUnavailableException {
+        synchronized (RestDatastoreClient.class) {
+            if (instance != null) {
+                if (esClientProvider != null) {
+                    logger.info("Stopping Elasticsearch rest client...");
+                    // all fine... try to cleanup the client
+                    cleanupClient(false);
+                    logger.info("Stopping Elasticsearch rest client... DONE");
+                } else {
+                    logger.warn("Close method called for a not initialized client!");
+                }
+            } else if (esClientProvider == null) {
+                // something wrong happened in the client lifecycle... try to cleanup and raise exception
+                cleanupClient(true);
+            }
+        }
+    }
+
+    private void cleanupClient(boolean raiseException) throws ClientUnavailableException {
+        Throwable cause = null;
+        try {
+            esClientProvider.getClient().close();
+            esClientProvider = null;
+            instance = null;
+        } catch (Throwable e) {
+            cause = e;
+            logger.error(CLIENT_CLEANUP_ERROR_MSG, e);
+        }
+        if (cause != null) {
+            throw new ClientUnavailableException(CLIENT_CLEANUP_ERROR_MSG, cause);
+        } else if (raiseException) {
+            throw new ClientUnavailableException(CLIENT_CLEANUP_ERROR_MSG);
+        }
+    }
+
+    @Override
+    public InsertResponse insert(InsertRequest insertRequest) throws ClientException {
+        checkClient();
+        try {
+            Map<String, Object> storableMap = modelContext.marshal(insertRequest.getStorable());
+            logger.debug("Insert - converted object: '{}'", storableMap.toString());
+            String json = MAPPER.writeValueAsString(storableMap);
+            HttpEntity entity = new NStringEntity(json, ContentType.APPLICATION_JSON);
+            Response insertResponse = esClientProvider.getClient().performRequest(
+                    POST_ACTION,
+                    getTypePath(insertRequest.getTypeDescriptor()),
+                    Collections.<String, String>emptyMap(), 
+                    entity);
+            if (isRequestSuccessful(insertResponse)) {
+                JsonNode responseNode = MAPPER.readTree(EntityUtils.toString(insertResponse.getEntity()));
+                String id = ((TextNode) responseNode.get(KEY_DOC_ID)).asText();
```

Comment:
Is casting required? 
JsonNode has `.asText()` so `esponseNode.get(KEY_DOC_ID).asText()` should work?
If confirmed please change all other usage with cast :)